app/src/main/kotlin/com/intive/picover/profile/view/ProfileUpdateBottomSheet.kt

```diff
@@ -82,24 +97,24 @@ fun ProfileUpdateBottomSheet(
 		) {
 			TextField(
 				modifier = Modifier.fillMaxWidth(),
-				value = userName,
+				value = userNameText,
 				onValueChange = {
-					updateUserName(it)
+					userNameText = it
 				},
 				supportingText = {
-					if (!validateName(userName)) {
+					if (!isUserNameValid) {
 						Text(
 							modifier = Modifier.fillMaxWidth(),
-							text = stringResource(id = getInvalidNameErrorMessageId(userName)),
+							text = stringResource(id = invalidNameErrorMessageId),
 						)
 					}
 					Text(
-						text = "${userName.length} / 20",
+						text = "${userNameText.length} / 20",
```

Comment:
Future improvement: Would be nice to connect somehow hardcoded value "20" with `maxLength` from used `TextValidator`. Nobody will remember about keeping these values up to date.